### PR TITLE
Remove minted_yCRV.

### PR DESCRIPTION
**Pull request by lychees**

### CODE REVIEW

File: src/UniDeposit.sol [hunk 1]
```diff
@@ -53,64 +55,68 @@ contract UniDeposit {
...
            mint();
         }
-        uint256 converted = get_yCrvFromUsdt(usdtBalance);
+        ycrvRequirement = get_yCrvFromUsdt(usdtBalance);
         // Will revert if there are not enough yCrv in contract though
-        bool isTransferOk = IERC20(yCrvToken).transfer(msg.sender, converted);
-        require(isTransferOk, "No enough yCrv in the contract. Please contract Dev team asap.");
+        bool isTransferOk = IERC20(yCrvToken).transfer(msg.sender, ycrvRequirement);
+        require(isTransferOk, "No enough yCrv in the contract. Please contract Dev team asap."); 
+        // minakokojima: should it failed inside the transfer call?
```
Comment (frankwei98) [inline src/UniDeposit.sol:119]: Maybe, but better safe than sorry.

File: src/UniDeposit.sol [hunk 2]
```diff
@@ -53,64 +55,68 @@ contract UniDeposit {
...
             // if enough, just swap then
-            uint256 yCrvWillGet = get_yCrvFromUsdt(amount);
-            IERC20(yCrvToken).transfer(msg.sender, yCrvWillGet);
+            uint256 output = get_yCrvFromUsdt(input);
+            mintedUSDT = mintedUSDT.sub(input);            
+            IERC20(yCrvToken).transfer(msg.sender, output);
         }
     }
 
+    function withdraw(uint input) external {
```
Comment (frankwei98) [inline src/UniDeposit.sol:88]: This one *is not* a bug though, but It makes me confused while I am reviewing.

I don't see this function have anything to do with `balance`.

So why this function was called `withdraw`?

If you take yCrv from `msg.sender` and send USDT back to him, shouldn't it called `swapToUsdt()` or something else?

Maybe we could have a `withdraw` function, for someone who deposited USDT already but needs to take back his USDT deposit before the next round of yCrv minting?

File: src/UniDeposit.sol [hunk 3]
```diff
@@ -53,64 +55,68 @@ contract UniDeposit {
...
+    }
+
+    // The world could always use more heroes.
     function mint() public {
-        uint256 yCrvBalanceBeforeMint = IERC20(yCrvToken).balanceOf(address(this));
-        // Minting now
-        // I don't know how to set this val
-        uint256 min_mint_amount = 0;
-
-        IUSDT(USDT).approve(yDeposit, unmintedUSDT);
```
Comment (frankwei98) [inline src/UniDeposit.sol:83]: You *should check* USDT allowance for `yDeposit` to spend USDT on behalf of `UniDeposit`, before addLiquidity to `yDeposit`. 
Even though you [approve `UINT_MAX` in the constructor](https://github.com/frankwei98/My-solidity-workshop/pull/1/files#diff-0afddb572d15ce58ef43d808d092aad3R40) already, there are chances that allowance is depleted.

File: src/UniDeposit.sol [hunk 4]
```diff
@@ -53,64 +55,68 @@ contract UniDeposit {
         _;
     }
 
+    function unminted_USDT() view public returns (uint) {
+        return IERC20(USDT).balanceOf(address(this));
```
Comment (frankwei98) [inline src/UniDeposit.sol:59]: Cool 👍

File: src/UniDeposit.sol (hunk 3 above)
Comment (frankwei98) [reply src/UniDeposit.sol:83]: I think this function needs to fix

**Review by frankwei98 — CHANGES_REQUESTED** [review]

File: src/UniDeposit.sol (hunk 2 above)
Comment (lychees) [reply src/UniDeposit.sol:88]: Withdraw() is yyCrv to USDT without 3% P3D.

File: src/UniDeposit.sol (hunk 3 above)
Comment (frankwei98) [reply src/UniDeposit.sol:83]: 🏳️

**Pull request merged**